tst/regression/CMakeLists.txt

```diff
@@ -0,0 +1,52 @@
+#=========================================================================================
+# (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+#
+# This program was produced under U.S. Government contract 89233218CNA000001 for Los
+# Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+# for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+# in the program are reserved by Triad National Security, LLC, and the U.S. Department
+# of Energy/National Nuclear Security Administration. The Government is granted for
+# itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+# license in this material to reproduce, prepare derivative works, distribute copies to
+# the public, perform publicly and display publicly, and to permit others to do so.
+#=========================================================================================
+
+# Adding regression tests
+#
+# To add additional regression tests to ctest, to things must be added to 
+# this file:
+# 
+# 1. there must exist within the regression/test_suites directory a folder associated with the test
+#   * For the calculate_pi example the folder is called 'calculate_pi', This
+#     folder must be added to the cmake variable TEST_DIRS as shown below. 
+
+# 2. the correct arguments to pass to the 'run_test.py' script must be specfied
+#   * This arguments must be added to the CMAKE variable TEST_ARGS as shown below
+#
+# Note each new test must append exactly 1 item to each of the CMAKE variables
+
+function(setup_test dir arg)
+  separate_arguments(arg) 
+  add_test( NAME regression_test:${dir} COMMAND python3 "${CMAKE_CURRENT_SOURCE_DIR}/run_test.py" ${arg} --test_dir "${CMAKE_CURRENT_SOURCE_DIR}/test_suites/${dir}")
+  set_tests_properties(regression_test:${dir} PROPERTIES LABELS "regression" )        
+endfunction()
+
+list(APPEND TEST_DIRS calculate_pi)
+list(APPEND TEST_ARGS "--driver ${CMAKE_BINARY_DIR}/example/calculate_pi/parthenon-example --driver_input ${CMAKE_CURRENT_SOURCE_DIR}/test_suites/calculate_pi/parthinput.regression")
```

Comment:
Some of the options, such as the mpi options, might be test system dependent. So I don't love that they're hardcoded here. Maybe we could add some system-dependent properties analogous to what @AndrewGaspar did in #114? Alternatively, `python` could parse options from some declarative machine database, rather than from the command line.

@brryan  and I did something along those lines in [nubhlight](https://github.com/lanl/nubhlight)

Reply:
Which file in that repo were you doing this?

Reply:
Compiler dependent variables/architecture dependent variables can be passed from cmake to the python script if there is a need for it.

Reply:
Seeing that CMake should probably know about these things in order to appropriately build the code, that would be my preference. 

Though I would need some help understanding the specifics of what needs to be passed in...?

Reply:
> Which file in that repo were you doing this?

The [config](https://github.com/lanl/nubhlight/blob/master/script/config.py) script reads in machine files from the [machine](https://github.com/lanl/nubhlight/tree/master/script/machine) directory and uses them to set parameters.

> Compiler dependent variables/architecture dependent variables can be passed from cmake to the python script if there is a need for it.

The big thing I'm thinking of is, for example, the `mpirun` command, which will not always be `mpirun -np N`.

Reply:
If it's possible to pass it into `cmake` from somewhere else, I'm happy with that solution. I just don't know very much about `cmake` so I don't think of it as a solution. :)

Reply:
So in cmake if MPI is found you have access to 
MPIEXEC_EXECUTABLE
    Executable for running MPI programs, if such exists.

Which can be used as a variable. But currently I'm dealing with a different problem with the ci. CI is crashing because it does not want me to run mpi as root which is the default for docker images.

Reply:
Ah, I see. Cool. In that case, we should provide logic in cmake for if mpi is not found. 

Calculate pi should work without mpi, so that's a solution for the short term... In the longer term, it's not great though. 

I think `openmpi` supports a flag `--allow-run-as-root` which should allow it to behave on docker? Not sure.

Reply:
Right now I have the regression test running without mpi. I'm guessing that it would be advantageous to have the same test running in serial and in parallel.

Reply:
Yes, it definitely would.

Reply:
This has been implemented.